extendr-api/src/ownership.rs

```diff
@@ -303,9 +318,12 @@ mod test {
                 let test_size = INITIAL_PRESERVATION_SIZE + EXTRA_PRESERVATION_SIZE * 5;
 
                 // Make some test objects.
-                let sexps = (0..test_size).map(|_| {
+                let sexp_pres = Rf_allocVector(VECSXP, test_size as R_xlen_t);
+                R_PreserveObject(sexp_pres);
```

Comment:
While this will work, it would be more idiomatic at this point to use `Rf_protect()`. You’re just protecting the `SEXP`, doing a quick calculation with it, and then release again.